libSetReplace/Match.cpp

```diff
@@ -220,8 +220,8 @@ class Matcher::Implementation {
       const auto token = Parallelism::acquire(Parallelism::HardwareType::StdCpu, rules_.size());
       const uint64_t& numThreadsToUse = token->numThreads();
 
-      auto addMatchesForRuleRange = [=](uint64_t start) {
-        for (uint64_t i = start; i < static_cast<uint64_t>(rules_.size()); i += numThreadsToUse) {
+      auto addMatchesForRuleRange = [=](RuleID start) {
+        for (RuleID i = start; i < static_cast<int>(rules_.size()); i += numThreadsToUse) {
```

Comment:
You should `static_cast` to `RuleID` rather than `int`. Casting to `int` only "works" because they happen to be equivalent.